File: llm_outputs/python_programs_corrected_by_llama/wrap.py
def wrap(text, cols):
    words = text.split()
    lines = []
    current_line = ""
    
    for word in words:
        if len(current_line) + len(word) + 1 <= cols:
            current_line += " " + word
        else:
            lines.append(current_line.strip())
            current_line = word + " "
            
    # Handle last line
    if current_line:
        lines.append(current_line.strip())

    return lines

File: llm_outputs/java_programs_corrected_by_llama/WRAP.java
package java_programs;

import java.util.ArrayList;

public class WRAP {

    public static void main(String[] args) {
        System.out.println("abc".indexOf('c', 30)); 
    }

    public static ArrayList<String> wrap(String text, int cols) {
        ArrayList<String> lines = new ArrayList<>();

        String line;
        while (!text.isEmpty()) { // changed condition
            int spaceIndex = text.indexOf(' ', cols); // found index for''
            if (spaceIndex == -1) {
                spaceIndex = text.length(); // default to end of string
            }
            line = text.substring(0, spaceIndex);
            text = text.substring(spaceIndex + 1); // updated substring call
            lines.add(line);
        }

        return lines;
    }
}
